cobigen/cobigen-core-parent/cobigen-core-test/src/main/java/com/devonfw/cobigen/test/assertj/GenerationReportToAssert.java

```diff
@@ -43,4 +43,43 @@ public GenerationReportToAssert isSuccessful() {
         }
         return this;
     }
+
+    /**
+     * Checks if an Exception of the given class was found in the {@link GenerationReportTo} error reports.
+     *
+     * @param exception
+     *            Class of Exception to check for
+     * @return {@link GenerationReportToAssert}
+     * @throws AssertionError
+     *             if another or no Exception was found in the report
+     */
+    public GenerationReportToAssert containsException(Class exception) throws AssertionError {
+
+        for (Throwable error : actual.getErrors()) {
+            if (error.getClass().equals(exception)) {
+                return this;
+            }
+        }
+        throw new AssertionError(
+            "Expected an Exception of class: " + exception + " in error reports but got something else.");
+    }
+
+    /**
+     * Checks if an Exception of the given class was NOT found in the {@link GenerationReportTo} error
+     * reports. errors.
+     *
+     * @param exception
+     *            Class of Exception to check for
+     * @return boolean
+     * @throws AssertionError
+     *             if the Exception was found in the report
+     */
+    public boolean notContainsException(Class exception) throws AssertionError {
```

Comment:
please return GenerationReportToAssert always to maintain builder pattern.

Reply:
same here regarding Class parameter type

Reply:
adjusted.